### ISSUE TITLE
make() fails when using decorate

### ISSUE THREAD

**Issue by tsteur** [PHP-DI/PHP-DI#554]
Hi @mnapoli 

I am getting currently this error with PHP-DI 5.4.5:

```
Entry "Piwik\Plugins\CoreHome\FooBar" cannot be resolved: Parameter $customer of __construct() has no value defined or guessableFull definition:Object ( class = Piwik\Plugins\CoreHome\FooBar scope = singleton lazy = false __construct( $customer = #UNDEFINED# ))
```

when calling:
```php
            return $container->make('Piwik\Plugins\CoreHome\FooBar', array(
                'customer' => $customer
            ));
```

and having a DI config like:

```php
'Piwik\Plugins\CoreHome\FooBar' => DI\decorate(function ($customer, \Interop\Container\ContainerInterface $c) {

        $myTest = $c->get('test.vars.myTest');
        if (!empty($myTest)) {
            return new MyTest();
        }

        return $customer;
    }),
```

The class looks like this:
```
class FooBar {
    /**
     * @var array
     */
    private $customer;

    public function __construct($customer)
    {
        $this->customer = $customer;
    }
```

It works nicely as soon as I remove the `DI\decorate()` but not when I add it. It does not even go into the decorator when debugging it.

It fails in `ObjectCreator::createInstance()` with a `DefinitionException`.

It seems like the passed parameters get lost in `DecoratorResolver::resolve(Definition $definition, array $parameters = [])` where the $parameters are actually never used.

**Comment by mnapoli** [comment]
I tried to push a commit to this branch but I am not allowed, here is the test I wrote:

```php
    /**
     * @see https://github.com/PHP-DI/PHP-DI/issues/554
     */
    public function testMakeWithDecorator()
    {
        $builder = new ContainerBuilder;
        $builder->addDefinitions([
            Foo::class => \DI\decorate(function ($previous) {
                return $previous;
            }),
        ]);
        $container = $builder->build();
        $result = $container->make(Foo::class, [
            'bar' => 'baz',
        ]);
        $this->assertEquals('baz', $result->bar);
    }
```

With this fixture class:

```php
namespace DI\Test\UnitTest\Fixtures;

class Foo
{
    public $bar;

    public function __construct($bar)
    {
        $this->bar = $bar;
    }
}
```

Let me know if that sounds like your problem (it does fail on 5.4).

**Comment by mnapoli** [comment]
I've released [5.4.6](https://github.com/PHP-DI/PHP-DI/releases/tag/5.4.6) which contains the fix.
